cmake/std/PullRequestLinuxDriverTest.py

```diff
@@ -42,706 +40,221 @@
-    parser.add_argument('targetBranch', action='store', help='Branch to merge to')
-
-    parser.add_argument('job_base_name',    help='The jenkins job base name')
-    parser.add_argument('github_pr_number', help='The github PR number')
-    parser.add_argument('job_number',       help='The jenkins build number')
-    parser.add_argument('workspaceDir',     help='The local workspace directory jenkins set up')
-
-    arguments = parser.parse_args()
-
-    return arguments
-
-
-def print_input_variables(arguments):
-    print("\n" + "="*90)
-    print("Jenkins Input Variables:")
-    print("- JOB_BASE_NAME: {job_base_name}".format(**vars(arguments)))
-    print("- WORKSPACE    : {workspaceDir}".format(**vars(arguments)))
-    print("\n" + "="*90)
-    print("Parameters:")
-    print("- TRILINOS_SOURCE_REPO  : {sourceRepo}".format(**vars(arguments)))
-    print("- TRILINOS_SOURCE_BRANCH: {sourceBranch}\n".format(**vars(arguments)))
-    print("- TRILINOS_TARGET_REPO  : {targetRepo}".format(**vars(arguments)))
-    print("- TRILINOS_TARGET_BRANCH: {targetBranch}\n".format(**vars(arguments)))
-    print("- PULLREQUESTNUM        : {github_pr_number}".format(**vars(arguments)))
-    print("- BUILD_NUMBER          : {job_number}".format(**vars(arguments)))
-    print("\n" + "="*90)
-
-
-def confirmGitVersion():
     """
-    Verify that the version of Git we're using is > 2.10
-    """
-    git_version_string = subprocess.check_output(['git', '--version'])
-    git_version_number_string = git_version_string[git_version_string.rfind(' '):]
-    major_git_version = int(git_version_number_string[:git_version_number_string.find('.')])
-    minor_git_version = int(git_version_number_string[git_version_number_string.find('.')+1:
-                                                      git_version_number_string.rfind('.')])
-
-    if major_git_version  <  2 or (major_git_version == 2 and minor_git_version < 10):
-        raise SystemExit("Git version  should be 2.10 or better - Exiting!")
-    else:
-        print(git_version_string)
-
-    return None
+    This function parses the command line arguments for the script.
 
-
-
-# TODO: moduleMap & environMap should be moved into files.
-#       Let's try YAML since it allows comments, etc.
-def setBuildEnviron(arguments):
-    """
-    TODO: Needs explanation.
-          - How is environMap structured?
-          - How is moduleMap structured?
-    TODO: wcm: Could it be possible to offload environMap and moduleMap into a
-          JSON or Yaml file?  (Yaml allows comments which is more helpful
-          than JSON IMO.)
+    Returns:
+        Namespace object containing the arguments from the command line.
     """
+    parser    = argparse.ArgumentParser(description='Parse the repo and build information', add_help=False)
+    required  = parser.add_argument_group('Required Arguments')
+    optional  = parser.add_argument_group('Optional Arguments')
+    cwd       = os.getcwd()
```

Comment:
Wherever possible, I'd recommend using `pathlib`'s `Path` in place of the equivalent functionality from `os`.

Reply:
Looks like this is Python3 only? 

We can look into this later... once our dependence on Python2 is gone.

```
[wcmclen@tr-test-4 ~]$ python
Python 2.7.9 (default, Feb 14 2020, 08:39:29)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named pathlib
```

Reply:
I need a "shaking my fist" emoji so I can shake my fist at Python 2.